lib/Doctrine/ORM/Internal/Hydration/IterableResult.php

```diff
@@ -26,7 +26,7 @@ class IterableResult implements \Iterator
     private $key = -1;
 
     /**
-     * @var object|null
+     * @var mixed[]|false|null
```

Comment:
`false` is not a type, it should be `bool` instead

Reply:
Counterpoint: `false` is absolutely a type, at least as far as PHP uses it. This one's backed up by phpDocumentor: https://docs.phpdoc.org/guides/types.html

Moreover, an understanding of `false` is necessary to properly check code like
```php
function takesInt(int $i) : void {}

function foo(string $s) : void {
  $a_pos = strpos($s, "a");
  
  if ($a_pos === false) {
    return;
  }
  
  takesInt($a_pos);
}
```

Reply:
> Counterpoint: false is absolutely a type

It's not a real _type_, type is _bool_; phpDocumentor describes it as _keyword_ that is allowed in type specification as a specialization for bool.
That code should really use array|null, not false - that is php4-ism just like most builtin functions (and that may be the reason why phpDocumentor allows it).

```php
function foo() : false {} // fatal error - https://3v4l.org/b9imQ
```

Reply:
> It's not a real type, type is bool;

You're right, I'm wrong.

> That code should really use `array|null`

Truly. `false` is an abomination. But it's one we're saddled with?